### PR TITLE
Organize imports quickfix action

### PR DESCRIPTION
**Pull request by kpodsiad**
Previously, our organize imports action supported only source.organizeImports kind (implemented in https://github.com/scalameta/metals/pull/1971) because code actions were filtered based on kind parameter (CodeActionProvider:50)
In order to show quickfix action vscode expects that kind of response is quickifx (hence in https://github.com/scalameta/metals/issues/2923 there was no action available although there was one in the response from the server but it was filtered out)

I extracted common logic to the abstract class and create a separate class per kind - IMO this is a good balance between DRY, single responsibility and code readability.   
Another option is to change `kind` in `CodeAction` to Set[String] and support multiple actions simultaneously but I'm not sure about this approach.  

I'll fix failing tests and implement new ones after receiving some feedback.

### CODE REVIEW

**Review by gabro — APPROVED** [review]
I personally like the approach and I think changing CodeAction to support multiple kinds would have been worse, since this case seems fairly unique.

**Comment by tgodzik** [comment]
I think it's a good approach, :+1: from me.

**Comment by ckipp01** [comment]
> Another option is to change kind in CodeAction to Set[String] and support multiple actions simultaneously but I'm not sure about this approach.

It's probably pretty rare that the same action would have multiple types. I'd maybe not do this unless you can think of another example. Either way,  no strong preference.

I like this being able to be offered as a QF action as well, so 👍🏼 from me.

**Comment by dos65** [comment]
IMO, it would be easier to keep the single instance of `OrganizeImports` and set its kind based on checking if the `CodeActionParams` has `"only": quickfix`.

Also, I'm wondering for which case we need this action with `source.organizeImports` kind ?

**Comment by ckipp01** [comment]
> Also, I'm wondering for which case we need this action with `source.organizeImports` kind ?

Most editors have an action built-in that a user can trigger a `Organize Imports` command that automatically calls the code action with the `source.organizeImports`. We should be sure to keep this in or else that functionality would break.

**Comment by dos65** [comment]
> Most editors have an action built-in that a user can trigger a Organize Imports command that automatically calls the code action with the source.organizeImports. We should be sure to keep this in or else that functionality would break.

Oh. Now I got what's going on. Then :+1: from me too

**Comment by kpodsiad** [comment]
It took me a while but I finally managed to write a few new tests. I'm not sure if `quickfix-available-on-hover` checks exactly what I wanted so please take a look at it

File: tests/unit/src/test/scala/tests/codeactions/OrganizeImportsLspSuite.scala [hunk 1]
```diff
@@ -170,9 +187,61 @@ class OrganizeImportsLspSuite
...
+  check(
+    "quickfix-available-on-hover",
+    """
+      |import scala.@@util.Try
+      |
+      |object A {
+      |  <<val action = "quick should be available">>
+      |}
+      |""".stripMargin,
+    s"${OrganizeImportsQuickFix.title}",
```
Comment (kpodsiad) [inline tests/unit/src/test/scala/tests/codeactions/OrganizeImportsLspSuite.scala:226]: Turned out that the test is failing 😅. Is there a way to mimic hover behavior in tests?

Comment (tgodzik) [reply tests/unit/src/test/scala/tests/codeactions/OrganizeImportsLspSuite.scala:226]: Quick fixes are unrelated to hovers, so no need to test them. I think the current test is `"on-unused"` is enough.

Comment (kpodsiad) [reply tests/unit/src/test/scala/tests/codeactions/OrganizeImportsLspSuite.scala:226]: Are you sure? In vscode when I hover unused there is quick fix action possible

File: metals/src/main/scala/scala/meta/internal/metals/codeactions/OrganizeImports.scala [hunk 2]
```diff
@@ -111,21 +158,9 @@ final class OrganizeImports(
...
 }
-object OrganizeImports {
-  def title: String = "Organize imports"
-  def kind: String = l.CodeActionKind.SourceOrganizeImports
 
-  type ScalaBinaryVersion = String
-  type ScalaVersion = String
+object OrganizeImportsQuickFix {
+  final val kind: String = l.CodeActionKind.QuickFix
+  final val title: String = "Fix organize imports"
```
Comment (tgodzik) [inline metals/src/main/scala/scala/meta/internal/metals/codeactions/OrganizeImports.scala:165]:
```suggestion
  final val title: String = "Fix imports"
```

**Review by tgodzik — CHANGES_REQUESTED** [review]
Thanks for the contribution! Looks good, but I would change the name slightly, what do you think?

**Comment by kpodsiad** [comment]
> Thanks for the contribution! Looks good, but I would change the name slightly, what do you think?

Yeah, I agree with that.  
BTW, I committed this suggestion from a browser, what a cool feature from Github!

**Review by tgodzik — APPROVED** [review]
LGTM!

**Pull request merged**